XO.py

```diff
@@ -0,0 +1,157 @@
+        if self.board[line][col] is not None:
+            return False
+        else:
+            self.board[line][col] = players[curr_player].symbol
+            return True
+
+    def line_won(self, players, curr_player):
+        for line in self.board:
+            if all(x == players[curr_player].symbol for x in line):
+                return True
+        return False
+
+    def col_won(self, players, curr_player):
+        for col in range(N):
+            if all(line[col] == players[curr_player].symbol for line in self.board):
+                return True
+        return False
+
+    def diagonal_won(self, players, curr_player):
+        # Main diagonal.
+        if all(self.board[index][index] == players[curr_player].symbol for index in range(N)):
+            return True
+        # Opposite diagonal.
+        if all(self.board[index][N-1-index] == players[curr_player].symbol for index in range(N)):
+            return True
+        return False
+
+
+class View:
+    def __init__(self):
+        pass
+
+    def print_board(self, board):
+        print()
+        print('Board:')
+        for line in board:
+            for item in line:
+                if item is None:
+                    print('-', end='')
+                else:
+                    print(item, end='')
+            print()
+        print()
+
+    def declare_winner(self, players, curr_player):
+        print()
+        print('***')
+        print(f'The winner is {players[curr_player].name}!!!')
+        print('***')
+
+    def declare_tie(self):
+        print()
+        print('***')
+        print('The result is a tie!!!')
+        print('***')
+
+
+class Controller:
+    def __init__(self, model, view):
+        self.model = model
+        self.view = view
+        self.players, self.curr_player = self.initiate_players()
+        self.number_of_moves_left = N * N  # Equals to number of squares on board.
+
+    def initiate_players(self):
+        Player = namedtuple('Player', 'name symbol')
+        x_player_name = input('Enter X player name\n')
+        o_player_name = input('Enter O player name\n')
+        while x_player_name == o_player_name:
+            o_player_name = input('Both names are the same. Choose a different name for O player\n')
+        players = [Player(name=x_player_name, symbol='x'), Player(name=o_player_name, symbol='o')]
+        curr_player = randint(0, 1)
+        return players, curr_player
+
+    def show_board(self):
+        board = self.model.get_board_status()
+        self.view.print_board(board)
+
+    def get_input(self):
+        line = self.is_input_valid('line')
+        col = self.is_input_valid('column')
+        updated = self.model.update_board(line, col, self.players, self.curr_player)
+        if not updated:
+            print(f'{self.players[self.curr_player].name}, you entered a taken spot on the board, try again')
+            return False
+        return True
+
+    def is_input_valid(self, input_type):

```

Comment:
Shouldn't this function be named - `get_valid_input` or something?
`is_X` functions are meant to verify something, not accept input from the user..

Reply:
Done